bolt/spark/array.py

```diff
@@ -467,6 +469,42 @@ def swapaxes(self, ax1, ax2):
 
         return self.transpose(p)
 
+    def reshape(self, *new):
+
+        new = argpack(new)
+        isreshapeable(new, self.shape)
+
+        if new == self.shape:
+            return self
+
+        i = self._simple_reshapeable(new)
+        if i == -1:
+            raise NotImplementedError("Currently no support for reshaping between keys and values for BoltArraySpark")
+        else:
+            new_key_shape, new_value_shape = new[:i], new[i:]
+            return self.keys.reshape(new_key_shape).values.reshape(new_value_shape)
+
+    def _simple_reshapeable(self, new):
```

Comment:
maybe rename to `_reshapebasic`? that would make it analogous to `_getbasic` and `_getadvanced`, which similarly breaks the core operation into two submethods depending on the complexity